dask/compose.py

```diff
@@ -0,0 +1,66 @@
+from uuid import uuid4
+import operator
+from functools import partial, wraps
+
+from toolz import merge
+
+from .optimize import cull
+from .context import _globals
+from . import threaded
+
+
+def get_uuid_dasks(v):
+    if isinstance(v, Value):
+        return v._uuid, v._dasks
+    else:
+        return v, []
```

Comment:
Hrm, although its not clear that you always want to traverse through large lists to see if they contain Values or not.